src/nomad_material_processing/vapor_deposition.py

```diff
@@ -0,0 +1,417 @@
+    measurement_type = Quantity(
+        type=MEnum(
+            "Heater thermocouple",
+            "Pyrometer",
+        )
+    )
+
+
+class Substrate(ArchiveSection):
+    m_def = Section(
+        a_plotly_graph_object={
+            "data": {"x": "temperature/duration", "y": "temperature/temperature"},
+            "layout": {"title": {"text": "Plotly Graph Object"}},
+            "label": "Plotly Graph Object",
+            "index": 1,
+        },
+    )
+    thin_film = Quantity(
+        description="""
+        The thin film that is being created during this step.
+        """,
+        type=ThinFilm,
+    )
+
+
+    # layers = SubSection(
+    #     description="""
+    #     An ordered list (starting at the substrate) of the thin films making up the
+    #     thin film stacks.
+    #     """,
+    #     section_def=ThinFilmReference,
+    #     repeats=True,
+    # )
+
+    temperature = SubSection(
+        section_def=SubstrateTemperature,
+    )
+    heater = Quantity(
+        type=MEnum(
+            "No heating",
+            "Halogen lamp",
+            "Filament",
+            "Resistive element",
+            "CO2 laser",
+        )
+    )
+    distance_to_source = Quantity(
+        type=float,
+        unit="meter",
+        description="""
+        The distance between the substrate and the source.
+        It is an arry because multiple sources can be used.
+        """,
+        shape=["*"],
+    )
+
+    substrate_specimen = SubSection(
+        description="""
+        The thin film stack that is being evaporated on.
+        """,
+        section_def=ThinFilmStackReference, 
+        #repeats=True, 
+    )
+
+
+class Pressure(ArchiveSection):
+    m_def = Section(
+        a_plot=dict(
+            x="process_time",
+            y="pressure",
+        ),
+    )
+    pressure = Quantity(
+        type=float,
+        unit="pascal",
+        shape=["*"],
+    )
+    process_time = Quantity(
+        type=float,
+        unit="second",
+        shape=["*"],
+    )
+
+
+class GasFlow(ArchiveSection):
+    gas = SubSection(
+        section_def=PureSubstanceSection,
+    )
+    flow = Quantity(
+        type=float,
+        unit="meter ** 3 / second",
+        shape=["*"],
+    )
+    process_time = Quantity(
+        type=float,
+        unit="second",
+        shape=["*"],  #################### why is this an array ?
+    )
+
+
+class ChamberEnvironment(ArchiveSection):
+    m_def = Section(
+        a_plot=dict(
+            x="pressure/process_time",
+            y="pressure/pressure",
+        ),
+    )
+    gas_flow = SubSection(
+        section_def=GasFlow,
+        repeats=True,
+    )
+    pressure = SubSection(
+        section_def=Pressure,
+    )
+
+
+class VaporDepositionStep(ActivityStep):
+    """
+    A step of any vapor deposition process.
+    """
+
+    m_def = Section()
+    creates_new_thin_film = Quantity(  ########## let's discuss together: how should the normalization work when we reprocess
```

Comment:
Yes. I will try to generalize the code from the `ikz_pld` plugin in a future PR.
```suggestion
    creates_new_thin_film = Quantity(
```